google/resource_compute_subnetwork.go

```diff
@@ -10,285 +24,403 @@ import (
 	"github.com/apparentlymart/go-cidr/cidr"
 	"github.com/hashicorp/terraform/helper/customdiff"
 	"github.com/hashicorp/terraform/helper/schema"
-	computeBeta "google.golang.org/api/compute/v0.beta"
-	"google.golang.org/api/compute/v1"
+	compute "google.golang.org/api/compute/v1"
 )
 
+// Whether the IP CIDR change shrinks the block.
+func isShrinkageIpCidr(old, new, _ interface{}) bool {
+	_, oldCidr, oldErr := net.ParseCIDR(old.(string))
+	_, newCidr, newErr := net.ParseCIDR(new.(string))
+
+	if oldErr != nil || newErr != nil {
+		// This should never happen. The ValidateFunc on the field ensures it.
+		return false
+	}
+
+	oldStart, oldEnd := cidr.AddressRange(oldCidr)
+
+	if newCidr.Contains(oldStart) && newCidr.Contains(oldEnd) {
+		// This is a CIDR range expansion, no need to ForceNew, we have an update method for it.
+		return false
+	}
+
+	return true
+}
+
+func splitSubnetID(id string) (region string, name string) {
```

Comment:
This seems to only be used for tests- mind moving it there?

Reply:
Kind of a hassle to move things between generated and non-generated code in one PR.  I can do it by copying the test into MagicModules, but that kind of doesn't feel right... is that how you'd suggest I do it?

Reply:
Ah yeah, you could open a quick PR in TF to do the move first, or we can just do it later.